### ISSUE TITLE
Incorrect handling of asset identifiers by usdchecker, usdcat

### ISSUE THREAD

**Issue by hinton-mgi** [PixarAnimationStudios/USD#1900]
### Description of Issue
I am just learning about USD, and was testing the usdchecker on some assets, including the kitchen set downloaded from [https://graphics.pixar.com/usd/release/dl_kitchen_set.html](https://graphics.pixar.com/usd/release/dl_kitchen_set.html). Both usdchecker and usdcat produce some strange results when run on Kitchen_set.usd. In particular, usdchecker reports unexpected "unresolvable external dependency" errors and flattening with usdcat appears to produce incorrect paths for the asset identifiers.

For reference, I have Kitchen_set.usd placed at `D:\temp\Kitchen_set\Kitchen_set.usd`.

Running `usdcat -o test.usda --flatten d:\temp\Kitchen_set\Kitchen_set.usd` produces a test.usda file where the asset identifier of the prim at `Kitchen_Set/Arch_grp/Kitchen_1` is `@d:/Temp/Kitchen_set/assets/Kitchen/assets/Kitchen/Kitchen.usd@` where I would have expected `@d:/Temp/Kitchen_set/assets/Kitchen/Kitchen.usd@`

Similarly, running `usdchecker d:\temp\Kitchen_set\Kitchen_set.usd` produces many errors, including `Found unresolvable external dependency 'd:/Temp/Kitchen_set/assets/Kitchen/assets/Kitchen/Kitchen.usd'. (fails 'MissingReferenceChecker')`. This path also has an extra "assets/Kitchen" in it and the error goes away if you edit Kitchen.usd and remove the asset identifier.

I don't even know what asset identifiers are for exactly, but to me this seems more likely to be a bug rather than an error in the asset. usdview has no trouble viewing it as far as I can tell, and does not produce any error or warning.

### Steps to Reproduce
1. Download and unzip the [Kitchen_set](https://graphics.pixar.com/usd/release/dl_kitchen_set.html)
2. With USD installed and/or on your path... run
3. `usdchecker Kitchen_set.usd` (outputs many unexpected unresolvable external dependency errors)
4. `usdcat -o test.usda --flatten Kitchen_set.usd` (asset identifiers in test.usda are incorrect)

### System Information (OS, Hardware)
Windows 10

### Package Versions
USD built from dev branch, HEAD at commit a470570c9492263ae8e973a91bd5800f998ef60b

### Build Flags
PXR_ENABLE_METAL_SUPPORT=FALSE
PXR_ENABLE_VULKAN_SUPPORT=FALSE
PXR_ENABLE_PTEX_SUPPORT=FALSE
PXR_USE_PYTHON_3=ON

**Comment by spiffmon** [comment]
Hi @hinton-mgi .  So, a full explanation of asset identifiers, their uses, and limitations gets into more advanced pipeline and ArResolver territory, which we probably at least should hint at [here](https://graphics.pixar.com/usd/release/glossary.html#usdglossary-assetinfo).

You are running into the limitations :-)  There are two ways around them, when using the ArDefaultResolver that ships with USD.  

The "easy but inconvenient" way is to be in the directory `d:\temp\Kitchen_set` when running those commands.

The "robust" way, that hints at the actual underlying issue and how you address it in general, is to provide some "configuration" for the resolver so that it knows how to resolve "relative, search-path" assetPaths like `@assets/Kitchen/Kitchen.usd@`.  Make sure the following environment variable is set in your shell/environment before running the commands:
`PXR_AR_DEFAULT_SEARCH_PATH = "d:\temp\Kitchen_set"`

HTH,
--spiff

**Comment by hinton-mgi** [comment]
Thanks @spiffmon, Was there some additional configuration to do besides setting `PXR_AR_DEFAULT_SEARCH_PATH`? Neither of these solutions work for either usdchecker or usdcat. I double checked that when `ArDefaultResolver()` is called, `envPath == "d:\temp\Kitchen_set"` on line 103 of `defaultResolver.cpp`.

**Comment by spiffmon** [comment]
Hi @hinton-mgi , I was finally able to poke at this a little.  I realized that the setup I described should not even be necessary, because by default, when you open a UsdStage on an asset using the DefaultResolver, it creates a Context that contains the directory in which the asset lives, i.e. the same thing we were specifying to PXR_AR_DEFAULT_SEARCH_PATH.

To my surprise, I saw the same behavior you did, and was _just_ about to conclude there was a bug somewhere, when I tried one more thing in usdview, and discovered the problem is not in the software, but in the asset, so `usdchecker` is actually telling us something true. :-) 

It looks like *some* (but not all, Kitchen.usd itself is fine) of the assets still have their `assetInfo["identifier"]` prefixed with a "./", which is incorrect, and causes the identifiers to be unresolvable at the stage-level.  I thought we had stripped it out, but apparently we haven't for the majority of the assets!

I may repurpose this report to indicate the problem - thanks for reporting!
